zds/utils/templatetags/interventions.py

```diff
@@ -78,10 +79,21 @@ def humane_delta(value):
 
 @register.filter('followed_topics')
 def followed_topics(user):
-    topics_followed = TopicAnswerSubscription.objects.filter(user=user,
-                                                             content_type__model='topic',
-                                                             is_active=True)\
-        .order_by('-last_notification__pubdate')[:10]
+    """
+    gets the ten last followed topics.
+    @TODO : make the "10" parameter a configuration one
+    :param user:
+    :return:
+    """
+    topics_followed = TopicAnswerSubscription.objects\
+                        .fetch_related("content_object",
+                                       "content_object__last_message")\
+                        .extra(sort_date=F("last_notification__pubdate"))\
+                        .extra(displayed_date=F("content_object__last_message__pubdate"))\
+                        .filter(user=user,
+                                content_type__model='topic',
+                                 is_active=True)\
```

Comment:
Non, et je n'arrive pas à le faire passer pour l'instant. Mais flake8 c'est
pas le plus important quand on joue avec ce genre de truc. Je le corrigerai
quand la fonctionnalité sera bonne.

2016-05-30 13:51 GMT+02:00 Gérard Paligot <notifications@github.com>:

> In zds/utils/templatetags/interventions.py
> <https://github.com/zestedesavoir/zds-site/pull/3643#discussion_r65061383>
> :
>
> > -                                                             is_active=True)\
> > -        .order_by('-last_notification__pubdate')[:10]
> > +    """
> > +    gets the ten last followed topics.
> > +    @TODO : make the "10" parameter a configuration one
> > +    :param user:
> > +    :return:
> > +    """
> > +    topics_followed = TopicAnswerSubscription.objects\
> > +                        .fetch_related("content_object",
> > +                                       "content_object__last_message")\
> > +                        .extra(sort_date=F("last_notification__pubdate"))\
> > +                        .extra(displayed_date=F("content_object__last_message__pubdate"))\
> > +                        .filter(user=user,
> > +                                content_type__model='topic',
> > +                                 is_active=True)\
>
> Ca passe flake8 ça ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zestedesavoir/zds-site/pull/3643/files/27c1f079d3238b57b64613e3be47dea44bf8aee4#r65061383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABc_xb9g4b6K4lHnWbRxWiiwGxgZZXACks5qGs81gaJpZM4Ipu2e>
> .
>